src/torchjd/sparse/_aten_function_overrides/shape.py

```diff
@@ -0,0 +1,289 @@
+    new_strides = t.strides[torch.tensor(dims)]
+    return StructuredSparseTensor(t.physical, new_strides)
+
+
+@impl(aten.cat.default)
+def cat_default(tensors: list[Tensor], dim: int) -> Tensor:
+    if any(not isinstance(t, StructuredSparseTensor) for t in tensors):
+        print_fallback(aten.cat.default, (tensors, dim), {})
+        return aten.cat.default([unwrap_to_dense(t) for t in tensors])
+
+    tensors_ = [cast(StructuredSparseTensor, t) for t in tensors]
+    ref_tensor = tensors_[0]
+    ref_strides = ref_tensor.strides
+    if any(not torch.equal(t.strides, ref_strides) for t in tensors_[1:]):
+        raise NotImplementedError(
+            "Override for aten.cat.default does not support SSTs that do not all have the same "
+            f"strides. Found the following tensors:\n{[t.debug_info() for t in tensors_]} and the "
+            f"following dim: {dim}."
+        )
+
+    # We need to try to find the (pretty sure it either does not exist or is unique) physical
+    # dimension that makes us only move on virtual dimension dim. It also needs to be such that
+    # traversing it entirely brings us exactly to the end of virtual dimension dim.
+
+    ref_virtual_dim_size = ref_tensor.shape[dim]
+    indices = torch.argwhere(
+        torch.eq(ref_strides[dim] * tensor(ref_tensor.physical.shape), ref_virtual_dim_size)
+        & torch.eq(ref_strides.sum(dim=0) * tensor(ref_tensor.physical.shape), ref_virtual_dim_size)
+    )
+    assert len(indices) <= 1
+
+    if len(indices) == 0:
+        # Add a physical dimension pdim on which we can concatenate the physicals such that this
+        # translates into a concatenation of the virtuals on virtual dimension dim.
+
+        pdim = ref_tensor.physical.ndim
+        physicals = [t.physical.unsqueeze(-1) for t in tensors_]
+        new_stride_column = torch.zeros(ref_tensor.ndim, 1, dtype=torch.int64)
+        new_stride_column[dim, 0] = ref_virtual_dim_size
+        new_strides = torch.concatenate([ref_tensor.strides, new_stride_column], dim=1)
+    else:
+        # Such a physical dimension already exists. Note that an alternative implementation would be
+        # to simply always add the physical dimension, and squash it if it ends up being not needed.
+        physicals = [t.physical for t in tensors_]
+        pdim = indices[0][0]
+        new_strides = ref_tensor.strides
+
+    new_physical = aten.cat.default(physicals, dim=pdim)
+    return StructuredSparseTensor(new_physical, new_strides)
+
+
+@impl(aten.expand.default)
+def expand_default(t: StructuredSparseTensor, sizes: list[int]) -> StructuredSparseTensor:
+    # note that sizes could also be just an int, or a torch.Size i think
+    assert isinstance(t, StructuredSparseTensor)
+    assert isinstance(sizes, list)
+    assert len(sizes) >= t.ndim
+
+    # Add as many dimensions as needed at the beginning of the tensor (as torch.expand works)
+    for _ in range(len(sizes) - t.ndim):
+        t = t.unsqueeze(0)
```

Comment:
Interesting: mypy is unhappy about this as `t.unsqueeze(0)` is a `Tensor`, not a `SST`. The thing is that we are supposed to use `unsqueeze_default` here. So something like this:

```suggestion
        t = unsqueeze_default(t, 0)
```
This unsqueeze of a sst happens at other places too and is inscribed in the idea of calling our operations whenever possible.